accepted/2021/system-drawing-win-only/system-drawing-win-only.md

```diff
@@ -0,0 +1,183 @@
+# Make `System.Drawing.Common` only supported on Windows
+
+**Owner** [Santiago Fernandez Madero](https://github.com/safern)
+
+In the .NET Core 2.x days we created the `Microsoft.Windows.Compatibility`
+package which exposes many Windows-only APIs so that more people can migrate to
+.NET Core. As part of this, we also ported `System.Drawing.Common`. This was
+mainly meant for .NET Core users that still target Windows, however since Mono
+already had a cross-platform implementation, we decided to bring that over. The
+cross-platform functionality is provided via the native library `libgdiplus`.
+Sadly, `libgdiplus` is a community supported library and at this point
+effectively unmaintained.
+
+While `libgdiplus` has helped a lot of customers expand to cross-platform it has
+also caused some headaches. `System.Drawing.Common` is very Windows driven as it
+is mostly a wrapper around GDI+ and is heavily coupled to Windows Forms, which
+keeps evolving and diverging from cross-platform graphics. This has caused some
+confusion over why some things don't work well on Unix or don't work not at all.
+
+Since the inclusion of the Mono cross-platform implementation we spent lot of
+time redirecting issues to `libgdiplus` that never got fixed, or helping people
+install `libgdiplus` correctly (because it's not part of .NET Core proper). We
+have thought about our options, and to support `System.Drawing.Common` properly
+on Unix we'd need to make `libgdiplus` a first-class citizen and ship it as part
+of the product:
```

Comment:
I think the first two issues are red-herrings. For example, the same things could be said about OpenSSL.

The difference with OpenSSL is that we're reacting to it, whereas we would need libgdiplus to react to us. Same comparison with ICU. That's the unworkable part. Also, the level of coupling is high.

Reply:
> I think the first two issues are red-herrings

It's a bit unclear what the two issues are you're referring to. Could you quote those?

The way I read the doc is that @safern meant to explain how the experience of using `libgdiplus` is very much subpar because it requires manual installation. If we wanted to make `System.Drawing.Common` work well, we'd need to invest to make this either much simpler or eliminate it entirely by bundling it. Neither are cheap.

Reply:
> Neither are cheap.

Nor worth the effort given `System.Drawing.Common` is pretty legacy and Winforms driven; and specially with the great managed libraries we have in the open source community already for xplat imaging/graphics.

Reply:
I guess that was Rich's argument: the real argument is that `System.Drawing.Common` for cross-platform use is a square peg in a round hole, no the complexities/cost around us making the experience great.

Reply:
Any suggestions to change the text here to express Rich's point?

Reply:
These were the two bullet points I was referencing. I agree that I wasn't clear.

> * `libgdiplus` doesn't ship with all operating systems by default, just like other native libraries we depend on (icu, openssl).
> * `libgdiplus` is community driven, so a lot of issues/PRs just go stale and are
  never looked at.

Reply:
OK, I've reworded the intro to remove the red herrings and focus on why we believe `libgdiplus` isn't a viable option.